rllib/agents/dqn/dqn.py

```diff
@@ -269,35 +289,16 @@ def update_target_if_needed(trainer, fetches):
         trainer.state["num_target_updates"] += 1
 
 
-def collect_metrics(trainer):
-    if trainer.config["per_worker_exploration"]:
-        # Only collect metrics from the third of workers with lowest eps
-        result = trainer.collect_metrics(
-            selected_workers=trainer.workers.remote_workers()[
-                -len(trainer.workers.remote_workers()) // 3:])
-    else:
-        result = trainer.collect_metrics()
-    return result
-
-
-def disable_exploration(trainer):
-    trainer.evaluation_workers.local_worker().foreach_trainable_policy(
-        lambda p, _: p.set_epsilon(0))
-
-
 GenericOffPolicyTrainer = build_trainer(
     name="GenericOffPolicyAlgorithm",
     default_policy=None,
     default_config=DEFAULT_CONFIG,
-    validate_config=check_config_and_setup_param_noise,
+    validate_config=validate_config_and_setup_param_noise,
     get_initial_state=get_initial_state,
-    make_policy_optimizer=make_optimizer,
-    before_init=setup_exploration,
-    before_train_step=update_worker_explorations,
-    after_optimizer_step=update_target_if_needed,
-    after_train_result=add_trainer_metrics,
-    collect_metrics_fn=collect_metrics,
-    before_evaluate_fn=disable_exploration)
+    make_policy_optimizer=make_policy_optimizer,
+    before_train_step=before_train_step,
+    after_optimizer_step=after_optimizer_step,
+    after_train_result=after_train_result)
```

Comment:
Please revert update_worker_exploration, update_target_if_needed, and setup_exploration to their original names, which were more explicit about what they did.

Reply:
Done. But setup_exploration does no longer setup the Exploration (DQN does not handle that anymore), so I took it out entirely.